autorest/utils/auth.go

```diff
@@ -26,6 +26,9 @@ import (
 // GetAuthorizer gets an Azure Service Principal authorizer.
 // This func assumes "AZURE_TENANT_ID", "AZURE_CLIENT_ID",
 // "AZURE_CLIENT_SECRET" are set as environment variables.
+//
+// Deprecated: Use GetAuthorizerFromClientCredentials from the
```

Comment:
Marking this as `Deprecated` following Go's convention (https://blog.golang.org/godoc-documenting-go-code)

Reply:
@jhendrixMSFT @marstr . Should we document the deprecation in the changelog ?

Reply:
Good question, it seems like it would be good to have a "Deprecated" section similar to how we have "New Features" and "Bugs" so that it's clear to folks what's been deprecated.